packages/hint-doctype/README.md

```diff
@@ -0,0 +1,106 @@
+# Check that the DOCTYPE tag of the document is valid
+
+This hint checks if the HTML is using the most modern DOCTYPE.
+
+## Why is this important?
+
+> In HTML, the doctype is the required "<!DOCTYPE html>" preamble
+found at the top of all documents. Its sole purpose is to prevent
+a browser from switching into so-called “quirks mode” when rendering
+a document; that is, the "<!DOCTYPE html>" doctype ensures that the
+browser makes a best-effort attempt at following the relevant specifications,
+rather than using a different rendering mode that is incompatible
+with some specifications.
+
+***From [MDN glossary for DOCTYPE specification][docmdn].***
+
+## What does the hint check?
+
+This hint checks if the HTML is using the most modern and valid DOCTYPE.
+
+It checks that the DOCTYPE is in the first line and that there are no other
+lines before the DOCTYPE. This is important as some browsers,
+including versions of IE prior to IE10, trigger quirks mode if a comment
+occurs before the DOCTYPE.
+
+It checks that there is no additional info apart from the DOCTYPE on the same
+line and that it is not duplicated elsewhere in the document. For compatibility
+with older browsers, it also accepts the legacy-compat DOCTYPE. Examples:
```

Comment:
The legacy-compat DOCTYPE is actually there to support older HTML content generators - not older browsers.